crates/pallet-feeds/src/lib.rs

```diff
@@ -17,29 +18,78 @@ pub mod pallet {
     pub struct Pallet<T>(_);
 
     pub type PutDataObject = Vec<u8>;
+    pub type FeedId = u64;
+
+    // TODO: make it more generic
+    #[derive(Encode, Decode, Debug, Clone, Eq, PartialEq)]
+    pub struct ObjectMetadata {
+        pub feed_id: FeedId,
+        pub hash: H256,
+        pub number: u32,
+    }
+
+    #[pallet::storage]
+    pub type Feeds<T: Config> = StorageMap<_, Blake2_128Concat, FeedId, (H256, u32), OptionQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn current_feed_id)]
+    pub type CurrentFeedId<T: Config> = StorageValue<_, FeedId, ValueQuery>;
 
     #[pallet::event]
     #[pallet::generate_deposit(pub(super) fn deposit_event)]
     pub enum Event<T: Config> {
-        DataSubmitted(PutDataObject, T::AccountId),
+        DataSubmitted(ObjectMetadata, T::AccountId),
+        FeedCreated(FeedId, T::AccountId),
     }
 
     #[pallet::error]
-    pub enum Error<T> {}
+    pub enum Error<T> {
+        UknownFeedId,
+    }
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
         // TODO: add proper weights
         #[pallet::weight(10_000)]
-        pub fn put(origin: OriginFor<T>, data: PutDataObject) -> DispatchResult {
+        pub fn put(
+            origin: OriginFor<T>,
+            data: PutDataObject,
+            metadata: ObjectMetadata,
+        ) -> DispatchResult {
             let who = ensure_signed(origin)?;
 
+            log::info!("metadata: {:?}", metadata);
+
+            let ObjectMetadata {
+                feed_id,
+                hash,
+                number,
+            } = metadata;
+
+            ensure!(Feeds::<T>::contains_key(feed_id), Error::<T>::UknownFeedId);
+
+            Feeds::<T>::mutate_exists(feed_id, |values| *values = Some((hash, number)));
+
             // TODO: add data handling
-            log::info!("SUBMITTED BY: {:?}", who);
-            log::info!("NEW DATA OBJECT: {:?}", data);
+            log::info!("data.len: {:?}", data.len());
+
+            Self::deposit_event(Event::DataSubmitted(metadata, who));
+
+            Ok(())
+        }
+
+        // TODO: add proper weights
+        #[pallet::weight(10_000)]
+        pub fn create_feed(origin: OriginFor<T>) -> DispatchResult {
```

Comment:
If above method is `put()` then this should be just `create()` for consistency, it makes sense considering this is all in pallet called `feeds`